src/full/recoder.rs

```diff
@@ -124,33 +136,38 @@ impl Recoder {
     /// Returns a `Vec<u8>` representing the new coded piece prepended with its
     /// source coding vector. The length of the returned vector is
     /// `self.get_full_coded_piece_byte_len()`.
-    pub fn recode<R: Rng + ?Sized>(&self, rng: &mut R) -> Vec<u8> {
-        let random_recoding_vector = (0..self.num_pieces_received).map(|_| rng.random()).collect::<Vec<u8>>();
+    pub fn recode<R: Rng + ?Sized>(&mut self, rng: &mut R, full_recoded_piece: &mut [u8]) -> Result<(), RLNCError> {
```

Comment:
The doc comment will need an update on `#Arguments` section mentioning `full_recoded_piece`.